src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String indexOutOfBoundNode = "While looking for Node"
+            + "Index is out of bound %d, Size %d";
+    private static final String indexOutOfBoundAdd = "While adding item"
+            + "we got Index is out of bound %d, Size %d";
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            tail(value);
```

Comment:
you see, your gradation and hierarchy of method calls in methods has gone the other way around because you are calling a more complex method inside a simpler one, when usually you descent from more abstract complex and vogue functionality to the smaller and simpler methods.
For example here it should have been like this
```suggestion
    public void add(T value) {
        // simpler logic than add by index
    }

    @Override
    public void add(T value, int index) {
        checkIndexForAdd(index);
        if (index == size) {
            add(value)
```